components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { ContextURL, GitpodServer, WorkspaceCreationResult } from "@gitpod/gitpod-protocol";
+import { SelectAccountPayload } from "@gitpod/gitpod-protocol/lib/auth";
+import { ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { Deferred } from "@gitpod/gitpod-protocol/lib/util/deferred";
+import { useCallback, useState } from "react";
+import { Redirect, useLocation } from "react-router";
+import Modal from "../components/Modal";
+import RepositoryFinder from "../components/RepositoryFinder";
+import SelectIDEComponent from "../components/SelectIDEComponent";
+import SelectWorkspaceClassComponent from "../components/SelectWorkspaceClassComponent";
+import { UsageLimitReachedModal } from "../components/UsageLimitReachedModal";
+import { openAuthorizeWindow } from "../provider-utils";
+import { getGitpodService, gitpodHostUrl } from "../service/service";
+import { LimitReachedOutOfHours, LimitReachedParallelWorkspacesModal } from "../start/CreateWorkspace";
+import { StartWorkspaceOptions } from "../start/start-workspace-options";
+import { StartWorkspaceError } from "../start/StartPage";
+import { useCurrentUser } from "../user-context";
+import { SelectAccountModal } from "../user-settings/SelectAccountModal";
+import Spinner from "../icons/Spinner.svg";
+import { useFeatureFlags } from "../contexts/FeatureFlagContext";
+
+export const useNewCreateWorkspacePage = () => {
+    const { startWithOptions } = useFeatureFlags();
+    const user = useCurrentUser();
+    return startWithOptions || user?.additionalData?.isMigratedToTeamOnlyAttribution;
```

Comment:
Curious on why we want to connect the user team-only migration to this new flow?

Reply:
User's who are in team only mode, will no longer have the cost center setting in the user settings. So it's even more important for them to be clear in which org the workspace is started.